src/java/org/apache/cassandra/service/accord/AccordSyncPropagator.java

```diff
@@ -232,25 +232,22 @@ public String toString()
                '}';
     }
 
-    public synchronized void onNodesRemoved(Set<Node.Id> removed)
+    public synchronized void onNodesRemoved(Node.Id removed)
     {
-        for (Node.Id node : removed)
+        PendingEpochs pendingEpochs = pending.get(removed.id);
+        if (pendingEpochs == null) return;
+        long[] toComplete = new long[pendingEpochs.size()];
+        Long2ObjectHashMap<PendingEpoch>.KeyIterator it = pendingEpochs.keySet().iterator();
+        for (int i = 0; it.hasNext(); i++)
+            toComplete[i] = it.nextLong();
+        Arrays.sort(toComplete);
+        for (long epoch : toComplete)
+            listener.onEndpointAck(removed, epoch);
```

Comment:
might be nice to move this and the below listener notifications outside of the synchronised block?